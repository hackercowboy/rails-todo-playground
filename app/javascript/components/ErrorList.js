import React from 'react';
import PropTypes from 'prop-types';
import ErrorListItem from './ErrorListItem';

const e = React.createElement;

function ErrorList({ errors }) {
  return errors ? e(
    'ul',
    { className: 'error-list' },
    errors.map((error) => e(ErrorListItem, { key: error, error })),
  ) : null;
}

ErrorList.propTypes = {
  errors: PropTypes.arrayOf(PropTypes.string),
};

ErrorList.defaultProps = {
  errors: undefined,
};

export default ErrorList;
