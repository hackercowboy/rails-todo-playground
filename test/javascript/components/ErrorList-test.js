import React from 'react';
import { shallow } from 'enzyme';

import ErrorList from '../../../app/javascript/components/ErrorList';

const e = React.createElement;

describe('<ErrorList/>', () => {
  it('should render without errors', () => {
    const wrapper = shallow(e(ErrorList));
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render with errors', () => {
    const wrapper = shallow(e(ErrorList, { errors: ['error1', 'error2'] }));
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
