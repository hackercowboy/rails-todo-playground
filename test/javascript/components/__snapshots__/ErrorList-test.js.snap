// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorList/> should render with errors 1`] = `
"<ul className=\\"error-list\\">
  <ErrorListItem error=\\"error1\\" />
  <ErrorListItem error=\\"error2\\" />
</ul>"
`;

exports[`<ErrorList/> should render without errors 1`] = `""`;
